### ISSUE TITLE
Broken usingRecursiveComparison assertion on java.util.regex.Pattern fields

### ISSUE THREAD

**Issue by alessandro-modolo** [assertj/assertj#2882]
**Broken usingRecursiveComparison assertion on java.util.regex.Pattern fields**
With the version 3.22.0, comparing two beans, that has a private field of type java.util.regex.Pattern, with the usingRecursiveComparison, was passing.

After the upgrade to version 3.23.0, the very same test is failing. I can't figure it out if it was a previous bug that has been fixed in the newer version, or if it is a regression that has been introduced.

* assertj core version: 3.23.0 and 3.23.1
* java version: 11.0.17
* test framework version: unit-jupiter 5.9.1
* os (if relevant): 

**Test case reproducing the bug**

This is the Bean class

```java
public class Bean {
    private final Pattern pattern;

    public Bean(String pattern) {
        this.pattern = Pattern.compile(pattern);
    }
}
```

and this is the test class

```java
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class BeanTest {
    @Test
    void patterTest() {
        // given
        String pattern = ".*";
        Bean bean1 = new Bean(pattern);
        // when
        Bean bean2 = new Bean(pattern);
        // then
        assertThat(bean2)
                .usingRecursiveComparison()
                .isEqualTo(bean1);
    }
}
```

**Comment by scordio** [comment]
> are you saying that there is was an inconsistent implementation inside the comparison strategy, that has been fixed in the newer version?

Yes, we decided to avoid the recursive comparison for JDK types and just rely on what `equals` does, even if some classes don't override it. This was done in 7406133f21ed66e3908a50c581ad0f0aaf059aa6, released in 3.23.0.

See also #2466 and #2450 for more background.

In light of the above, I am closing this issue.

**Comment by joel-costigliola** [comment]
@alessandro-modolo if you implements `equals`, you would need to register it with `withEqualsForType` since the recursive comparison ignores all non jdk type `equals` method (unless you call `usingOverriddenEquals()` but that will enable all it for all types)
